packages/replay-internal/src/util/createPerformanceEntries.ts

```diff
@@ -141,28 +141,78 @@ function createResourceEntry(
 }
 
 /**
- * Add a LCP event to the replay based on an LCP metric.
+ * Add a LCP event to the replay based on a LCP metric.
  */
 export function getLargestContentfulPaint(metric: {
   value: number;
+  rating: 'good' | 'needs-improvement' | 'poor';
   entries: PerformanceEntry[];
-}): ReplayPerformanceEntry<LargestContentfulPaintData> {
+}): ReplayPerformanceEntry<WebVitalData> {
+  return getWebVital(metric, 'largest-contentful-paint');
+}
+
+/**
+ * Add a CLS event to the replay based on a CLS metric.
+ */
+export function getCumulativeLayoutShift(metric: {
+  value: number;
+  rating: 'good' | 'needs-improvement' | 'poor';
+  entries: PerformanceEntry[];
+}): ReplayPerformanceEntry<WebVitalData> {
+  return getWebVital(metric, 'cumulative-layout-shift');
+}
+
+/**
+ * Add a FID event to the replay based on a FID metric.
+ */
+export function getFirstInputDelay(metric: {
+  value: number;
+  rating: 'good' | 'needs-improvement' | 'poor';
+  entries: PerformanceEntry[];
+}): ReplayPerformanceEntry<WebVitalData> {
+  return getWebVital(metric, 'first-input-delay');
+}
+
+/**
+ * Add an INP event to the replay based on an INP metric.
+ */
+export function getInteractionToNextPaint(metric: {
+  value: number;
+  rating: 'good' | 'needs-improvement' | 'poor';
+  entries: PerformanceEntry[];
+}): ReplayPerformanceEntry<WebVitalData> {
+  return getWebVital(metric, 'interaction-to-next-paint');
+}
+
+/**
+ * Add an web vital event to the replay based on the web vital metric.
+ */
+export function getWebVital(
+  metric: {
+    value: number;
+    rating: 'good' | 'needs-improvement' | 'poor';
+    entries: PerformanceEntry[];
+  },
+  name: string,
+): ReplayPerformanceEntry<WebVitalData> {
```

Comment:
Wondering if it's better to use this function instead of having all of the wrappers.